### PR TITLE
PSR-1319 Fixed QA bug

### PR DESCRIPTION
**Pull request by westwater**

### CODE REVIEW

File: app/transformations/MemberPaymentsTransformer.scala [hunk 1]
```diff
@@ -156,35 +157,46 @@ class MemberPaymentsTransformer @Inject()(
...
+            case None =>
+              currentMemberDetail.copy(memberPSRVersion = None)
+            case Some(initialMemberDetail) =>
+              val normalisedInitialMember = normalise(initialMemberDetail)
+              val normalisedCurrentMember = normalise(currentMemberDetail)
+              val same = normalisedInitialMember == normalisedCurrentMember
+              if (!same) {
+                logger.info(s"member $index has changed, removing memberPSRVersion")
+                if(logger.isDebugEnabled) {
+                  logger.debug(Diff(normalisedInitialMember, normalisedCurrentMember).mkString(" - "))
```
Comment (westwater) [inline app/transformations/MemberPaymentsTransformer.scala:193]: prints human readable, field specific diffs - great for debugging issues in the future around recordVersion / memberPsrVersion

File: app/models/requests/psr/MemberPayments.scala [hunk 2]
```diff
@@ -75,7 +75,11 @@ case class MemberLumpSumReceived(
   lumpSumAmount: Double,
   designatedPensionAmount: Double
 ) {
-  val zero: Boolean = lumpSumAmount == 0.0 && designatedPensionAmount == 0.0
+  val zero: Boolean = lumpSumAmount == 0d && designatedPensionAmount == 0d
```
Comment (kberec) [inline app/models/requests/psr/MemberPayments.scala:78]: Would it be useful to differentiate in naming between this `zero` and the `zero` below? I.e. rename this to `isZero` maybe?

File: app/transformations/MemberPaymentsTransformer.scala [hunk 3]
```diff
@@ -156,35 +157,46 @@ class MemberPaymentsTransformer @Inject()(
         }
       }
 
+      val normalise = (memberDetails: MemberDetails) => {
```
Comment (kberec) [inline app/transformations/MemberPaymentsTransformer.scala:160]: If we change the task list not to distinguish between In progress and Completed, do we need to revert this change? (FYI @ccatosdevelopment )

File: scalastyle-config.xml [hunk 4]
```diff
@@ -87,7 +87,7 @@
     </check>
     <check level="warning" class="org.scalastyle.scalariform.CyclomaticComplexityChecker" enabled="true">
         <parameters>
-            <parameter name="maximum"><![CDATA[30]]></parameter>
+            <parameter name="maximum"><![CDATA[35]]></parameter>
```
Comment (kberec) [inline scalastyle-config.xml:90]: Even 30 is quite hight TBH....

File: app/transformations/MemberPaymentsTransformer.scala (hunk 3 above)
Comment (westwater) [reply app/transformations/MemberPaymentsTransformer.scala:160]: Yeah we would, this is just to fix the QA bug

**Review by kberec — APPROVED** [review]

File: scalastyle-config.xml (hunk 4 above)
Comment (westwater) [reply scalastyle-config.xml:90]: Yeah it is but it's just the nature of the transformer - might be worth splitting out all the code into seperate methods and the core `toEtmp/fromEtmp` is just a simple for-comp
